packages/babel-generator/src/generators/statements.js

```diff
@@ -123,10 +123,14 @@ function buildLabelStatement(prefix, key = "label") {
     const label = node[key];
     if (label) {
       this.space();
-
-      const terminatorState = this.startTerminatorless();
-      this.print(label, node);
-      this.endTerminatorless(terminatorState);
+      if (prefix == "break" || prefix == "continue") {
+        // see https://git.io/vQ7uG
+        this.print(label, node);
```

Comment:
I realise that one @loganfsmyth, you also mentioned it before. Having parens next to `break` and `continue` is wrong (?), so I wanted to get that out of the way.